presto-hive/src/main/java/io/prestosql/plugin/hive/metastore/SemiTransactionalHiveMetastore.java

```diff
@@ -2116,9 +2166,22 @@ private static void renameDirectory(HdfsContext context, HdfsEnvironment hdfsEnv
         runWhenPathDoesntExist.run();
 
         try {
-            if (!hdfsEnvironment.getFileSystem(context, source).rename(source, target)) {
+            FileSystem fileSystem = hdfsEnvironment.getFileSystem(context, source);
+            if (!fileSystem.rename(source, target)) {
                 throw new PrestoException(HIVE_FILESYSTEM_ERROR, format("Failed to rename %s to %s: rename returned false", source, target));
             }
+            // setOwner failure shouldn't cause sql failure
```

Comment:
If this was explicitly requested, we should probably not hide this problem, why would we want that?

I am thinking, however, we could want to have this enabled by default.
@electrum @willmostly @alexjo2144 wdyt?